### PR TITLE
#101390216 Request Refund

### PR DESCRIPTION
**Pull request by andela-fsenjobi**
#### What does this PR do?
Users who have paid for tickets to an event that is cancelled can request refunds
#### Description of Task to be completed?
Add feature test for request refunds
Add and run migration for refunds
Include route to request refund
Modify test to accomodate changes
Hide attend button for cancelled events
Create and modify view
Resolve permission conflicts
Use decorators to clean up view
Allow only event managers to manage their events
Use `cancancan` for access control
#### How should this be manually tested?
Become an event manager and create an event with a paid ticket
Add a user account to attend the event by buying a ticket
Log in as an Event manager and cancel the event by clicking disable
Log in as a User, you should see a button to request a refund
#### Also included
#119349579 Event Managers can only manage their own events (bg)
#119542077 Users cannot attend cancelled Events (bg)
#### Screenshots
image
image

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class BookingsController < ApplicationController
-  before_action :authenticate_user,
-                except: [:paypal_hook, :paypal_dummy]
+  # load_and_authorize_resource param_method: :request_refund
```
Comment (andela-sdamian) [inline app/controllers/bookings_controller.rb]: remove the comment

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -19,7 +18,7 @@ def event_titles
   end
 
   def index
-    @bookings = current_user.user_tickets
+    @bookings = current_user.bookings.order(id: :desc).decorate
```
Comment (andela-sdamian) [inline app/controllers/bookings_controller.rb]: Do you mind taking the order method to a scope in the model

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -56,6 +51,18 @@ def scan_ticket
     flash[:notice] = "Ticket does not exist" unless @user_ticket
   end
 
+  def request_refund
+    @booking = Booking.find_by_uniq_id(params[:uniq_id])
+    if @booking.update_attributes(
```
Comment (andela-sdamian) [inline app/controllers/bookings_controller.rb]: What `houdci-bot`  is saying here is that this `request_refund` method can be refactored into 
```

def request_refund 
  @booking = Booking.find_by_uniq_id(params[:uniq_id])

  flash[:notice] = if @booking.update_attributes( refund_requested: true,
                      time_requested: Time.now)
                      "Request for a refund has been sent"
                   else
                      "Request cannot be sent at this time."
                   end

end 

```

File: app/views/bookings/request_refund.js.erb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+location.reload();
+<% if flash[:notice].present? %>
+  Materialize.toast("<%= flash[:notice].to_s %>", 3000, "rounded")
```
Comment (andela-sdamian) [inline app/views/bookings/request_refund.js.erb]: I think we need to discuss on having just one `Materialize.toast()` method in our entire application.

Comment (andela-fsenjobi) [inline app/views/bookings/request_refund.js.erb]: That would be great

**Pull request merged**